src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,47 @@
 'use strict';
 
+const healPoints = require('./healPoints');
+
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.name = name;
+    this.health = healPoints(100);
+    Animal.alive.push(this);
+  }
+
+  static removeDiedAmimals() {
+    Animal.alive = Animal.alive.filter(aliveAnimal => aliveAnimal.health > 0);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name, damage = 50) {
+    super(name);
+    this.damage = damage;
+  }
```

Comment:
I do not see in the conditions of the task that the damage of a predator can be different. Let's do everything according to the conditions
```suggestion

```